### PR TITLE
Added a support to show children categories count through the getById Rest API

### PR DESCRIPTION
**Pull request by hassan-mustafa-baig**
### Proposed Changes
* A rest API upgraded to accept showChildrenCount parameter. Upon providing **showChildrenCount=true** it will expose child categories count in the API response that will be visible in the **childrenCount** property but in case of **showChildrenCount=false** the property **childrenCount** will be excluded from the Rest API response

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:

image

image

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java [hunk 1]
```diff
@@ -61,6 +66,29 @@ public CategoryView toCategoryView(final Category category, final User user) {
                 .sortOrder(category.getSortOrder())
                 .categoryVelocityVarName(category.getCategoryVelocityVarName())
                 .build();
+
+
+        return Response.ok(new ResponseEntityView(categoryView)).build();
+    }
+
+    public Response toCategoryWithChildCountView(final Category category, final User user) throws DotDataException, DotSecurityException {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java:74]: Missing Javadoc

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryWithChildCountView.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
+package com.dotcms.rest.api.v1.categories;
+
+import java.util.Date;
+
+public class CategoryWithChildCountView {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryWithChildCountView.java:5]: Missing Javadoc

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java [hunk 3]
```diff
@@ -61,6 +66,29 @@ public CategoryView toCategoryView(final Category category, final User user) {
                 .sortOrder(category.getSortOrder())
                 .categoryVelocityVarName(category.getCategoryVelocityVarName())
                 .build();
+
+
+        return Response.ok(new ResponseEntityView(categoryView)).build();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java:71]: you need to use a sub class if use Response as a return type, so swagger may do the doc

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java [hunk 4]
```diff
@@ -61,6 +66,29 @@ public CategoryView toCategoryView(final Category category, final User user) {
...
+    }
+
+    public Response toCategoryWithChildCountView(final Category category, final User user) throws DotDataException, DotSecurityException {
+
+        final Integer childrenCategoriesCount = this.categoryAPI.findChildren(user, category.getInode(),
+                    false, "").size();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java:77]: Use StringPool.BLANK

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java [hunk 5]
```diff
@@ -61,6 +66,29 @@ public CategoryView toCategoryView(final Category category, final User user) {
...
+                .keywords(category.getKeywords())
+                .key(category.getKey())
+                .categoryName(category.getCategoryName())
+                .active(category.isActive())
+                .sortOrder(category.getSortOrder())
+                .categoryVelocityVarName(category.getCategoryVelocityVarName())
+                .childrenCount(childrenCategoriesCount)
+                .build();
+
+        return Response.ok(new ResponseEntityView(categoryWithChildCountView)).build();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java:91]: Same here

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoriesResource.java [hunk 6]
```diff
@@ -316,7 +317,9 @@ public final CategoryView getCategoryByIdOrKey(@Context final HttpServletRequest
             throw new DoesNotExistException(
                     "Category with idOrKey: " + idOrKey + " does not exist");
         }
-        return this.categoryHelper.toCategoryView(category, user);
+
+        return showChildrenCount ? Response.ok(new ResponseEntityView(this.categoryHelper.toCategoryWithChildCountView(category, user))).build() :
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoriesResource.java:321]: still the same, need to use ResponseEntityView<> at least

**Review by jdotcms — APPROVED** [review]

**Pull request merged**